app/extensions/string_extensions.rb

```diff
@@ -363,39 +365,46 @@ class String
       "nbsp"  => " "
     }.freeze
   end
+  # rubocop:enable Layout/AlignHash
   # :startdoc:
 
   # This should safely match anything that could possibly be interpreted as
   # an HTML tag.
   HTML_TAG_PATTERN = %r{</*[A-Za-z][^>]*>}.freeze
 
   ### Textile-related methods ###
-  #
+
   def t(sanitize = true)
-    Textile.textilize_without_paragraph(self, false, sanitize).html_safe
+    Textile.textilize_without_paragraph_safe(self, false, sanitize)
   end
 
   def tl(sanitize = true)
-    Textile.textilize_without_paragraph(self, true, sanitize).html_safe
+    Textile.textilize_without_paragraph_safe(self, true, sanitize)
   end
 
-  # TODO: Move somewhere that content_tag is defined
+  # Textilize string, wrapped in a <div>, making it all safe for output
   def tp(sanitize = true)
-    '<div class="textile">'.html_safe + Textile.textilize(self, false, sanitize).html_safe + "</div>".html_safe
+    textile_div_safe { Textile.textilize(self, false, sanitize) }
   end
 
+  # Textilize string (with links), wrapped in a <div>,
+  # making it all safe for output
   def tpl(sanitize = true)
-    '<div class="textile">'.html_safe +
-      Textile.textilize(self, true, sanitize).html_safe +
-      "</div>".html_safe
+    textile_div_safe { Textile.textilize(self, true, sanitize) }
+  end
+
+  def textile_div_safe
+    %(<div class="textile">#{yield}</div>).
```

Comment:
Seems like this should go into textile.rb or maybe if there was a textile_helper.rb it could get access to content_tag.

Reply:
I'll give both of them a try. 
I've been wondering how to both give it access to content_tag, yet have it accessible to everything that now calls it.

Reply:
Yeah, since starting to do a lot more Rails at work, I haven't gotten used to Rail's opinions about what is available where or figured out how to work around them consistently.  Python is very explicit about this sort of thing which I'm getting more convinced is the right way to go.

Reply:
I moved `textile_div_safe` to Textile, and added a test.  I couldn't find a a way to provide access - as indicated above - to `content_tag`.

Reply:
Yeah, I think to resolve it we might have to move these methods to no longer be on String and into a proper helper.  However, that would mean changing all calls from something like `s.tp` to `tp s` which would be a real pain for very little gain.